assignments/arrays.js

```diff
@@ -63,34 +63,75 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
 
 // ==== Challenge 1 ====
 // The dealer can't recall the information for a car with an id of 33 on his lot. Help the dealer find out which car has an id of 33 by logging the car's year, make, and model in the console log provided to you below:
-console.log(`Car 33 is a *car year goes here* *car make goes here* *car model goes here*` );
+function getInventoryCar(carId) {
+  for (i = 0; i < inventory.length; i++){
+    if (inventory[i].id === carId){
+      return inventory[i];
+    }
+  }
 
+  return "car ID: " + carId + " not found";
+}
+console.log(`Car 33 is a ` + getInventoryCar(33).car_year + ' ' + getInventoryCar(33).car_make + ' ' + getInventoryCar(33).car_model);
 
+//// Stretch use map()
+inventory.map(function (car){
+  if (car.id === 33) {
+    console.log(`(stretch map()): Car 33 is a ` + car.car_year.toString() + ' ' + car.car_make + ' ' + car.car_model);
+  }
+});
 
 // ==== Challenge 2 ====
 // The dealer needs the information on the last car in their inventory.  What is the make and model of the last car in the inventory?  Log the make and model into the console.
-let lastCar = 0;
-console.log();
+let lastCar = inventory.length - 1;
+console.log("Last car make and model: " + inventory[lastCar].car_make + ' ' + inventory[lastCar].car_model);
 
 // ==== Challenge 3 ====
 // The marketing team wants the car models listed alphabetically on the website. Sort all the car model names into alphabetical order and log the results in the console
 let carModels = [];
-console.log();
+for (let i = 0; i < inventory.length; i++){
+  carModels.push(inventory[i].car_model);
+}
+carModels = carModels.sort();
+console.log(carModels);
 
 // ==== Challenge 4 ====
 // The accounting team needs all the years from every car on the lot. Create a new array from the dealer data containing only the car years and log the result in the console.
 let carYears = [];
-console.log();
+for (let i = 0; i < inventory.length; i++){
+  carYears.push(inventory[i].car_year);
+}
+console.log(carYears);
 
 // ==== Challenge 5 ====
 // The car lot manager needs to find out how many cars are older than the year 2000. Using the carYears array you just created, find out how many cars were made before the year 2000 by populating the array oldCars and logging it's length.
-let oldCars =[];
-console.log(); 
+let oldCars = [];
+for (let i = 0; i < carYears.length; i++){
+  if (carYears[i] < 2000) {
+    oldCars.push(carYears[i]);
+  }
+}
+console.log(oldCars.length + ' cars are older than 2000');
+
+//// Stretch use reduce()
+let totalNumOldCars = 0;
+carYears.reduce(function (totalOldCars, carYear) {
```

Comment:
I don't know if I'd use reduce here instead of filter. But if you want to use reduce you'd want to do something like this 
- `let count = inventory.reduce(function(n, val) {
    return n + (val.car_year < 2000);
}, 0);`